test/e2e/e2ev1.go

```diff
@@ -30,77 +29,62 @@ import (
 
 var _ = Describe("kubebuilder", func() {
 	Context("with v1 scaffolding", func() {
-		imageName := "controller:v0.0.1"
 		var testSuffix string
-		var c *config
-		var kbTest *kubebuilderTest
+		var kbc *KbContext
 
 		BeforeEach(func() {
 			var err error
 			testSuffix, err = randomSuffix()
 			Expect(err).NotTo(HaveOccurred())
-			c, err = configWithSuffix(testSuffix)
+			env := []string{"GO111MODULE=off"}
+			kbc, err = ContextWithSuffix(testSuffix, env)
 			Expect(err).NotTo(HaveOccurred())
-			kbTest = &kubebuilderTest{
-				Dir: c.workDir,
-				Env: []string{"GO111MODULE=off"},
-			}
-			prepare(c.workDir)
+			Expect(kbc.Preparewd()).To(Succeed())
 		})
 
 		AfterEach(func() {
 			By("clean up created API objects during test process")
-			resources, err := kbTest.RunKustomizeCommand("build", filepath.Join("config", "default"))
-			if err != nil {
-				fmt.Fprintf(GinkgoWriter, "error when running kustomize build during cleaning up: %v\n", err)
-			}
-			if _, err = kbTest.RunKubectlCommandWithInput(resources, "delete", "--recursive", "-f", "-"); err != nil {
-				fmt.Fprintf(GinkgoWriter, "error when running kubectl delete during cleaning up: %v\n", err)
-			}
-			if _, err = kbTest.RunKubectlCommand(
+			kbc.KustomizeCleanup(filepath.Join("config", "default"))
+			if _, err := kbc.Kubectl.Command(
 				"delete", "--recursive",
 				"-f", filepath.Join("config", "crds"),
 			); err != nil {
 				fmt.Fprintf(GinkgoWriter, "error when running kubectl delete during cleaning up crd: %v\n", err)
 			}
 
-			By("remove container image created during test")
-			kbTest.CleanupImage(c.controllerImageName)
-
-			By("remove test work dir")
-			os.RemoveAll(c.workDir)
+			By("remove container image and work dir")
+			kbc.Destroy()
```

Comment:
I am wondering if we should move `KustomizeCleanup` and "kubectl delete recursive" also in the Destroy or cleanup() method ?

Reply:
The cleanup tasks are different between v1 and v2.

In v1, we need to
1. cleanup manifests that installed through kustomize.
1. cleanup CRD, since CRDs are installed separately.
1. rm image and work dir.

In v2, we need to
1. cleanup manifests that installed through kustomize. CRDs are deleted in this step.
1. uninstall cert-manager bundle.
1. rm image and work dir.

IMO it's better to leave the cleanup steps as is.

Reply:
Ack